doc/api.md

```diff
@@ -1178,6 +1188,183 @@ In this case a batch update (`POST /v2/op/update`) to CB will be generated with
+
+{
+  "value": "Ping request",
+  "type": "command"
+}
+
+```
+
+It is important to note that parameter `type`, with the entity type must be included.
+
+Context Broker API is quite flexible and allows to update an attribute in several ways. Please have a look to the [Orion API]([http://telefonicaid.github.io/fiware-orion/api/v2/stable](https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/orion-api.md)) for details.
+
+**Important note**: don't use operations in the NGSI API with creation semantics. Otherwise, the entity/attribute will be created locally to Context Broker and the command will not progress to the device (and you will need to delete the created entity/attribute if you want to make it to work again). Thus, the following operations *must not* be used:
+
+* `POST /v2/entities`
+* `POST /v2/entities/<id>/attrs`
+* `PUT /v2/entities/<id>/attrs`
+* `POST /v2/op/entites` with `actionType` `append`, `appendStrict` or `replace`
+
+### Command reception
+
+Once the command is triggered, it is send to the device. Depending on transport protocol, it is going to be sent to the device in a different way. After sending the command, the IoT Agent will have updated the value of `ping_status` to `PENDING` for entity into the Context Broker. Neither
+`ping_info` nor `ping` will be updated.
+
+#### HTTP devices
+
+**Push commands**
+
+Push commands are those that are sent to the device once the IoT Agent receives the request from the Context Broker. In order to 
+send push commands it is needed to set the `"endpoint": "http://[DEVICE_IP]:[PORT]/"` in the device or group provision. The device 
+is supposed to be listening for commands at that URL in a synchronous way. Make sure the device endpoint is reachable by the IoT 
+Agent. Push commands are only valid for HTTP devices. For MQTT devices it is not needed to set the `endpoint` parameter. 
+
+Considering using the IoTA-JSON Agent, and given the previous example, the device should receive a POST request to 
+`http://[DEVICE_IP]:[PORT]` with the following payload:
+
+```
+POST /
+Content-Type: application/json
+
+{"ping":"Ping request"}
+```
+
+**Poll commands**
+
+Poll commands are those that are stored in the IoT Agent waiting to be retrieved by the devices. This kind of 
+commands are typically used for devices that doesn't have a public IP or the IP cannot be reached because of 
+power or netkork constrictions. The device connects to the IoT Agent periodically to retrieve commands. In order 
+to configure the device as poll commands you just need to avoid the usage of `endpoint` parameter in the device provision.
+
+Once the command request is issued to the IoT agent, the command is stored waiting to be retrieved by the device. In that moment, the status of the command is `"<command>_status": "PENDING"`. 
+
+For HTTP devices, in order to retrieve a poll command, the need to make a GET request to the IoT Agent to the path `/iot/json` with the following parameters:
```

Comment:
Is this path hardwired? Or it depends on some parametrization?

Reply:
The path depends of the resource definition, but it seems have some issues using a different one that the default (see https://github.com/telefonicaid/iotagent-node-lib/issues/1524)

Reply:
Assuming that it works (i.e. assuming that #1524 get fully solved), `resource` can be used only at group provisioning? or it is also in the API of device provisioning?

Reply:
Checking in api.md:

* [Group provisioning](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#config-group-datamodel): includes `resource`
* [Device provisioning](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#device-datamodel) doesn't include `resource`

So it seems it can be configured only at group level. @AlvaroVega could you confirm?

Reply:
> Checking in api.md:
> 
>     * [Group provisioning](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#config-group-datamodel): includes `resource`
> 
>     * [Device provisioning](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#device-datamodel) doesn't include `resource`
> 
> 
> So it seems it can be configured only at group level. @AlvaroVega could you confirm?

Yes it is. But resource is still a leftover in device mongo model.

Reply:
Based in the feedback above, I'd suggest the following modification:

```suggestion
For HTTP devices, in order to retrieve a poll command, the need to make a GET request to the IoT Agent to the path used as `resource` in the provisioned group (`/iot/json` by default if no `resource` is used) with the following parameters:

FIXME [#1524](https://github.com/telefonicaid/iotagent-node-lib/issues/1524) `resource` different to `/iot/json` is not working at the present moment, but it will when this issue gets solved.
```

Reply:
Added here 99eb5163b979168971b85ef4e649d014cc6164de